controller/instance_manager_controller.go

```diff
@@ -1702,16 +1704,39 @@ func (imc *InstanceManagerController) deleteOrphans(im *longhorn.InstanceManager
 	return nil
 }
 
-func (imc *InstanceManagerController) isOrphanDeletable(orphan *longhorn.Orphan, autoDelete bool, instanceMap instanceProcessMap, isInstanceScheduledOnNode func(node, instance string) (bool, error)) (isDeletable bool, err error) {
-	if autoDelete {
-		return true, nil
+func (imc *InstanceManagerController) deleteOrphan(orphan *longhorn.Orphan, resourceDeletable bool) error {
+	if resourceDeletable {
+		orphan.Spec.Parameters[longhorn.OrphanResourceDeletable] = strconv.FormatBool(resourceDeletable)
+		_, err := imc.ds.UpdateOrphan(orphan)
+		switch {
+		case datastore.ErrorIsNotFound(err):
+			return nil
+		case err != nil:
+			return errors.Wrapf(err, "failed to update %s on orphan %v before deletion", longhorn.OrphanResourceDeletable, orphan.Name)
+		}
 	}
 
+	err := imc.ds.DeleteOrphan(orphan.Name)
+	switch {
+	case datastore.ErrorIsNotFound(err):
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to delete orphan %v", orphan.Name)
+	}
+	return nil
+}
+
+func (imc *InstanceManagerController) isOrphanDeletable(orphan *longhorn.Orphan, autoDelete bool, instanceMap instanceProcessMap, isInstanceScheduledOnNode func(node, instance string) (bool, error)) (isDeletable, isResourceDeletable bool, err error) {
 	instanceName := orphan.Spec.Parameters[longhorn.OrphanInstanceName]
 	instanceProc, instanceExist := instanceMap[instanceName]
+	isResourceDeletable = instanceExist
+	if autoDelete {
+		return true, isResourceDeletable, nil
+	}
```

Comment:
Why not continue the following checks for autoDelete?

Reply:
`autoDelete` means that we can initiate the orphan CR deletion directly, without checking the other CRs. Will rename the method `isOrphanDeletable` to `isOrphanCrDeletable`.

Reply:
But why not check the other CRs?

Reply:
The `autoDelete` parameter is enabled when:

1. instance manager is terminating, or
2. the `orphan-resource-auto-deletion` is enabled

In this two situation, we can initiate the orphan CR deletion directly, and orphan controller will take care about the resource clearning. After this flag check, it checks whether the corresponding instance is scheduled back to the node so that it is no longer considered to be orphaned.

Reply:
Got your point. the resource deletable should also consider the resource relocation even when auto delete is enabled.